src/adapters/axelar/AxelarReceiverAdapter.sol

```diff
@@ -0,0 +1,123 @@
+import "../../libraries/Error.sol";
+import "../../libraries/Types.sol";
+import "../../libraries/Message.sol";
+
+import "./interfaces/IAxelarGateway.sol";
+import "./interfaces/IAxelarExecutable.sol";
+import "./libraries/StringAddressConversion.sol";
+
+/// @notice receiver adapter for axelar bridge
+contract AxelarReceiverAdapter is IAxelarExecutable, IBridgeReceiverAdapter {
+    using StringAddressConversion for string;
+
+    IAxelarGateway public immutable gateway;
+    IGAC public immutable gac;
+
+    /*/////////////////////////////////////////////////////////////////
+                        STATE VARIABLES
+    ////////////////////////////////////////////////////////////////*/
+    address public senderAdapter;
+    string public senderChain;
+
+    mapping(bytes32 => bool) public isMessageExecuted;
+    mapping(bytes32 => bool) public commandIdStatus;
+
+    mapping(string => uint256) public reverseChainIdMap;
+
+    /*/////////////////////////////////////////////////////////////////
+                                 MODIFIER
+    ////////////////////////////////////////////////////////////////*/
+    modifier onlyCaller() {
+        if (!gac.isPrevilagedCaller(msg.sender)) {
+            revert Error.INVALID_PREVILAGED_CALLER();
+        }
+        _;
+    }
+
+    /*/////////////////////////////////////////////////////////////////
+                         CONSTRUCTOR
+    ////////////////////////////////////////////////////////////////*/
+    constructor(address _gateway, address _gac) {
+        gateway = IAxelarGateway(_gateway);
+        gac = IGAC(_gac);
+    }
+
+    /*/////////////////////////////////////////////////////////////////
+                         EXTERNAL FUNCTIONS
+    ////////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc IBridgeReceiverAdapter
+    function updateSenderAdapter(bytes memory _senderChain, address _senderAdapter) external override onlyCaller {
+        string memory _senderChainDecoded = abi.decode(_senderChain, (string));
+
+        if (keccak256(abi.encode(_senderChainDecoded)) == keccak256(abi.encode(""))) {
+            revert Error.ZERO_CHAIN_ID();
+        }
+
+        if (_senderAdapter == address(0)) {
+            revert Error.ZERO_ADDRESS_INPUT();
+        }
+
+        address oldAdapter = senderAdapter;
+        senderAdapter = _senderAdapter;
+        senderChain = _senderChainDecoded;
+
+        emit SenderAdapterUpdated(oldAdapter, _senderAdapter, _senderChain);
+    }
+
+    /// @dev accepts new cross-chain messages from axelar gateway
+    function execute(
+        bytes32 commandId,
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload
+    ) external override {
+        /// @dev step-1: validate incoming chain id
+        if (keccak256(bytes(sourceChain)) != keccak256(bytes(senderChain))) {
+            revert Error.INVALID_SENDER_CHAIN_ID();
+        }
```

Comment:
nit: `sourceChain` here is technically the chain name and not the chain id right? Should we update the comment and error message accordingly?